runhouse/resources/hardware/cluster.py

```diff
@@ -275,6 +276,8 @@ def config(self, condensed=True):
             config["ssl_certfile"] = self.cert_config.cert_path
             config["ssl_keyfile"] = self.cert_config.key_path
 
+        cpu_or_cuda = detect_cuda_version_or_cpu(self)
+        config["process_unit"] = cpu_or_cuda.upper() if cpu_or_cuda == "cpu" else "GPU"
```

Comment:
This isn't really what the config is for - I'd rather just cache this propery within the ClusterServelet than add it to the config. "processing_unit" also isn't exactly an accurate descriptor - A GPU box also has CPUs, and there are different kinds of GPUs. I'd just call it "has_cuda" for now.